datajunction-clients/python/datajunction/_base.py

```diff
@@ -65,7 +65,7 @@ def serialize_item(item):
                         if isinstance(item, dict)
                         else item
                     )
-                except TypeError:  # Ignore and try the next candidate
+                except (TypeError, AttributeError):  # Ignore and try the next candidate
```

Comment:
Maybe we should try to raise a clearer error here as well, as in, have an additional exception case that catches broad exceptions and adds a better error message.

Reply:
I think this was pretty easy to see already, but unfortunately very hard to reproduce.